java/core/src/main/java/co/worklytics/psoxy/gateway/StorageEventRequest.java

```diff
@@ -3,26 +3,33 @@
 import lombok.Builder;
 import lombok.Getter;
 import lombok.NonNull;
+import lombok.With;
 
 import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
 
 /**
  * Request received when a change is done in some storage service, such as "new" object created or "updated" object
  */
+@With
 @Builder
 @Getter
 public class StorageEventRequest {
     /**
      * Stream where the object can be read
      */
-
-    @NonNull
     InputStreamReader readerStream;
 
     /**
-     * Name of the bucket where the object is. Example: "mybucket"
+     * Stream to write output to
      */
+    OutputStreamWriter destinationStream;
```

Comment:
OK, but it is kinda weird that this "request" class contains readers & writers, it's very attached to the implementation. But can't provide a better approach.
passing around IO is always tricky if closed/flushed when it is not supposed to happen

Reply:
yeah, it should be split into two things